### ISSUE TITLE
[Bug] 1.21.4 Block stacking incorrectly. 

### ISSUE THREAD

**Issue by MikeCokos** [MarkusBordihn/BOs-Easy-Mob-Farm#103]
I'm using version 8.8.0 on 1.21.4. Any block that gets mined has a normal tag, but the nbt data changes on player pickup. So new items do not stack with the edited items already in the player's inventory.

**Comment by MarkusBordihn** [comment]
Thank you for the report! I attempted to reproduce the issue, but I wasn’t able to observe the same behavior on my side.

From my testing, a dirt block obtained using the `/give` command has the same data as one that is mined. 

Could you please provide additional details to help clarify the issue? For example, try holding the two different blocks in your hand and run the following command:

```
/data get entity @s SelectedItem
```

This will display the item data for each block. Sharing the output or a screenshot would be very helpful.

In general, the mod does not modify any standard blocks, so I might not fully understand the issue you are describing. Your additional input will help me investigate this further.

**Comment by MikeCokos** [comment]
It seems on my end that the tags are added once in the inventory. So using /give is useless. You need to check the item while it's on the ground and before it enters an inventory of any kind. Then pickup the item and check it again. Best way to get the item is to go in survival mode and mine dirt

**Comment by MikeCokos** [comment]
![{7DBCAA01-D9F7-4535-A337-EAEF960C61F4}](https://github.com/user-attachments/assets/f2934881-cd2a-4923-b4b4-db536e962ce3)

![{4A6A74A4-01C3-4658-B666-2D6ECAA480C6}](https://github.com/user-attachments/assets/87f5dc8e-bcec-4cd8-bf78-52c9ddcc6d02)

**Comment by MarkusBordihn** [comment]
Thank you for providing the details! That’s interesting, normally, a `minecraft:sand` block does not have entity data when placed in the world, as it’s just a block. 
However, in your case, the block appears to carry entity data even outside your inventory.

This suggests that a mod on your end might be converting normal `minecraft:sand` blocks into entities to add additional functionality.  

A quick search for `"frozenSpottingIconTicksToCheck"` indicates it might be related to **FrozenLib**. 
You can see more here:  https://github.com/search?q=frozenSpottingIconTicksToCheck&type=code

Could you confirm if you are using any mods that rely on **FrozenLib** on your end?

**Comment by MikeCokos** [comment]
I am not using any mods that use FrozenLib.

**Comment by MarkusBordihn** [comment]
Thanks for the confirmation. Unfortunately, I’m still unable to reproduce the issue on my end.  :(

At this point, I suspect an **incompatibility with another mod** may be causing the problem, as my mod does not modify the inventory or the items within it in any way.  

Additionally, version **1.21.4** no longer uses any custom models like the previous versions, so I’m unsure what might be happening in your case.

In the meantime, I have uploaded an updated version **8.9.0** with a few minor bug fixes. 
While I’m not certain they address this particular issue, there’s a chance it may already be resolved.  

To investigate this further, could you please provide the following details:  
- **Mod Loader** you are using (e.g., Fabric, Forge, NeoForge).  
- A **list of mods** (or a link to the modpack on CurseForge or Modrinth) you are using.  
- If possible, a **short video** demonstrating the issue.  

This information will help me reproduce and debug the issue more effectively.

**Comment by MikeCokos** [comment]
I'm using Fabric. 
![{DD186A54-3096-40CA-9D09-8FAF3B145B47}](https://github.com/user-attachments/assets/e9c7d4ce-aef3-4405-83eb-a586ed86b850)

https://github.com/user-attachments/assets/d465a6bf-d32b-4c40-91e3-a97620039cd9

**Comment by MarkusBordihn** [comment]
Thank you for the additional details. I was able to narrow down the issue to the following mod combination:

![image](https://github.com/user-attachments/assets/474e6502-a168-4817-9aca-b5bb14b6052b)

However, I’m currently unsure why empty Mob Card data are being injected into the blocks. 
My assumption is that one of the mods might be automatically adding the `minecraft:custom_data` to the blocks.

To address this, I’ll try defining my own unique custom_data, which will hopefully prevent it from being injected into the blocks moving forward.

**Comment by rivyu123** [comment]
Have the same exact issue! I have trailer tales as well, could it be that?

**Comment by MarkusBordihn** [comment]
I encountered the same issue by the combination of three specific mods.

If you're able to reproduce the issue with "Trailer Tales" alone, and it disappears when that mod is disabled, it would be helpful to confirm. 
That way, I can focus specifically on Trailer Tales to see if there's a potential fix on my end.

Please keep in mind that Minecraft 1.21.4 is still in beta for modding, so some issues with mod compatibility and implementation are to be expected. 
I appreciate your patience and any additional feedback you can provide!

**Comment by maugwei** [comment]
I have the same problem but never had "Trailer Tales" I did have "Polymer" But after removal the bug persists. I submitted my own ticket but thought I might as well mention it here also. I also posted my modlist in my submission ticket.

**Comment by MarkusBordihn** [comment]
I’ve fixed the issue by creating a custom Mob Capture Card data component. 
It took a few days of testing and fixing some related crashes.  

However, this change means **existing Mob Capture Cards will no longer be valid** because their data won’t match the new format. Unfortunately, I wasn’t able to automatically convert the old cards.  

The fix prevents the bug since other mods can no longer directly access this data and unintentionally apply it to other blocks or entities.  

That said, similar issues could still occur with **other mods and their data components.** 
In such cases, the **3rd-party mod developer** would need to address the problem.  

As a reminder, **mod support for 1.20.4 is still in beta,** so these types of issues should hopefully get ironed out once a stable release arrives. Thanks for your patience and understanding! 😊

**Comment by MikeCokos** [comment]
Nice fix. Thanks for taking time to really solve this, or atleast work around the mod that's breaking it. Works well in my testing.

**Comment by superdupermario9** [comment]
Any plans to port this fix to 1.21.1? I would love to use this mod in my current pack but the fix is only available for 1.21.4.

**Comment by MarkusBordihn** [comment]
Yes, I'm currently working on backporting the fix to 1.21.3, and after that, to 1.21.1. 🚧

Unfortunately, it will take a bit of time since I can't just copy the changes directly.
There are significant differences between these versions, and each one requires specific adjustments.  😅
